### PR TITLE
Ruby Capstone Project

### PR DESCRIPTION
**Pull request by kaizipaul**
### Classes and methods that were implemented:
- Book and Label Classes, and their methods (@kaizipaul)
- MusicAlbum and Genre Classes, and their methods (@DJ-MrJay)
- Games and Author Classes, and their methods (@AbayomiOlaoye)
- Storing data for Books and Labels (@kaizipaul)
- Storing data for Games and Authors (@AbayomiOlaoye)
- Storing data for MusicAlbum and Genre (@DJ-MrJay)

###  >>[Video Link](https://drive.google.com/uc?id=1pm0qmmgij-ggmfk4Sy2JOwFYLtJHElHO&export=download)<< to the demo.

### CODE REVIEW

File: schema/schema.sql [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  publish_date DATE NOT NULL,
+  archived BOOLEAN NOT NULL
+);
+
+-- Labels
+CREATE TABLE labels (
+  id SERIAL PRIMARY KEY,
+  title VARCHAR(150) NOT NULL,
+  color VARCHAR(150) NOT NULL
+);
```
Comment (muneebulrehman) [inline schema/schema.sql:63]: Kindly have a reference to the author, genre,  source and label tabels from books, music and games table. I have recorded [this video](https://www.loom.com/share/3bb2fbd21a324177b1ba95212895e1bc) where I explained in detail what you need to fix.

File: src/book.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+require_relative './item'
+
+class Book < Item
```
Comment (muneebulrehman) [inline src/book.rb:3]: Every book should have an author, genre, and label. However, you are missing all of these things. I have recorded [this video](https://www.loom.com/share/3bb2fbd21a324177b1ba95212895e1bc) where I explained in detail what you need to fix.

File: src/game.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+require 'date'
+require_relative './item'
+
+class Game < Item
```
Comment (muneebulrehman) [inline src/game.rb:4]: Every game should have an author, genre and label. However, you are missing all of these things. I have recorded [this video](https://www.loom.com/share/3bb2fbd21a324177b1ba95212895e1bc) where I explained in detail what you need to fix.

File: src/musicalbum.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
+require_relative './app'
+require_relative './item'
+require 'date'
+
+class MusicAlbum < Item
```
Comment (muneebulrehman) [inline src/musicalbum.rb:5]: Every music album should have an author, genre and label. However, you are missing all of these things. I have recorded [this video](https://www.loom.com/share/3bb2fbd21a324177b1ba95212895e1bc) where I explained in detail what you need to fix.

File: src/genre.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+require_relative './item'
+require 'date'
+
+class Genre < Item
```
Comment (muneebulrehman) [inline src/genre.rb:4]: There should be an option to display all the created genres in the UI but your application seems to miss it. Kindly add this functionality. Every book should have an author, genre, source, and label. However, you are missing all of these things. I have recorded [this video](https://www.loom.com/share/3bb2fbd21a324177b1ba95212895e1bc) where I explained in detail what you need to fix.

File: src/author.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
+require 'date'
+require_relative './item'
+
+class Author < Item
```
Comment (muneebulrehman) [inline src/author.rb:4]: There should be an option to display all the created authors in the UI but your application seems to miss it. Kindly add this functionality. Every book should have an author, genre, source, and label. However, you are missing all of these things. I have recorded [this video](https://www.loom.com/share/3bb2fbd21a324177b1ba95212895e1bc) where I explained in detail what you need to fix.

File: src/label.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+require_relative './item'
+
+class Label < Item
```
Comment (muneebulrehman) [inline src/label.rb:3]: There should be an option to display all the created labels in the UI but your application seems to miss it. Kindly add this functionality. Every book should have an author, genre, source, and label. However, you are missing all of these things. I have recorded [this video](https://www.loom.com/share/3bb2fbd21a324177b1ba95212895e1bc) where I explained in detail what you need to fix.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR

_**Note: I mistakenly mentioned implementing the `source` class. However, it is in the requirement of the group of 4. Kindly ignore where ever I have mentioned implementing the `source` class.**_

## Required Changes ♻️
_Check the comments under the review._

- [x] The video link you submitted is private. Kindly make it public so the next reviewer can review it. 
image

- [x] Kindly watch [this video](https://www.loom.com/share/3bb2fbd21a324177b1ba95212895e1bc) where I tried to demonstrate what requirements you still need to implement.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/app.rb [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    @label_list = []
+  end
+
+  def music_display
+    list_music
+  end
+
+  def genre_display
+    list_genres
+  end
```
Comment (rotimiazeez) [inline src/app.rb:25]: - Please note that I get some errors when I try to list genres. Kindly fix it. 👍🏽
Screenshot from 2023-01-27 21-18-15

File: src/app.rb [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+    show_authors
+  end
+
+  def book_display
+    book_list
+  end
+
+  def label_display
+    label_list
+  end
```
Comment (rotimiazeez) [inline src/app.rb:49]: - Please note that I get errors when I try to list labels. Kindly fix it. 👍🏽
Screenshot from 2023-01-27 21-17-02

File: src/app.rb [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+    show_games
+  end
+
+  def game_create
+    create_game
+  end
+
+  def author_display
+    show_authors
+  end
```
Comment (rotimiazeez) [inline src/app.rb]: - Please note that I get errors when I try to list authors. Could you make sure this method is implemented? 👍🏽
Screenshot from 2023-01-27 21-16-21

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
Hi TEAM 😉, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## To highlight
- Good Teamwork ✔️
- Good test implementation ✔️
- Git flow followed ✔️
- No linter error ✔️

## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something needs to be 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by sumairq — APPROVED** [review]
### Hi Team, 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**